blazar/db/migration/alembic_migrations/versions/02e2f2186d98_extra_capability.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2014 OpenStack Foundation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""extra capability
+
+Revision ID: 02e2f2186d98
+Revises: 520c9976941c
+Create Date: 2020-04-17 15:51:40.542459
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '02e2f2186d98'
+down_revision = '520c9976941c'
+
+from itertools import chain
+import uuid
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import mysql
+
+
+def upgrade():
+    op.create_table('extra_capabilities',
+                    sa.Column('id', sa.String(length=36), nullable=False),
+                    sa.Column('resource_type', sa.String(255), nullable=False),
+                    sa.Column('capability_name', sa.String(255),
+                              nullable=False),
+                    sa.Column('private', sa.Boolean, nullable=False,
+                              server_default=sa.false()),
+                    sa.Column('created_at', sa.DateTime(), nullable=True),
+                    sa.Column('updated_at', sa.DateTime(), nullable=True),
+                    sa.PrimaryKeyConstraint('id'),
+                    sa.UniqueConstraint('resource_type', 'capability_name'))
+
+    if op.get_bind().engine.name != 'sqlite':
+        connection = op.get_bind()
+
+        host_query = connection.execute("""
```

Comment:
Just curious, how did you generate this file? This seems like a tricky migration to do.

Reply:
I did the auto-generate then did manual edits. The auto-generation didn't handle migrating any data so that was all custom.